libs/ledgerjs/packages/hw-app-btc/src/BtcNew.ts

```diff
@@ -453,3 +443,57 @@ function accountTypeFromArg(
   if (arg.segwit) return new p2wpkhWrapped(psbt, masterFp);
   return new p2pkh(psbt, masterFp);
 }
+
+/*
+  The new protocol only allows standard path.
+  Standard paths are (currently):
+  M/44'/(1|0)'/X'
+  M/49'/(1|0)'/X'
+  M/84'/(1|0)'/X'
+  M/86'/(1|0)'/X'
+  M/48'/(1|0)'/X'/Y'
+  followed by "", "(0|1)", or "(0|1)/b", where a and b are 
+  non-hardened. For example, the following paths are standard
+  M/48'/1'/99'/7'
+  M/86'/1'/99'/0
+  M/48'/0'/99'/7'/1/17
+  The following paths are non-standard
+  M/48'/0'/99'           // Not deepest hardened path
+  M/48'/0'/99'/7'/1/17/2 // Too many non-hardened derivation steps
+  M/199'/0'/1'/0/88      // Not a known purpose 199
+  M/86'/1'/99'/2         // Change path item must be 0 or 1
+*/
+function isPathNormal(path: string): boolean {
+  //path is not deepest hardened node of a standard path or deeper, use BtcOld
+  const h = 0x80000000;
+  const pathElems = pathStringToArray(path);
+
+  const hard = (n: number) => n >= h;
+  const soft = (n: number | undefined) => !n || n < h;
+  const change = (n: number | undefined) => !n || n == 0 || n == 1;
```

Comment:
if `n === 0` then `!n` will also return `true` so the condition is useless

Reply:
fixed